compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -2119,6 +2108,41 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         diag
     }
 
+    fn emit_tuple_wrap_err(
+        &self,
+        err: &mut DiagnosticBuilder<'tcx>,
+        span: Span,
+        found: Ty<'tcx>,
+        expected: Ty<'tcx>,
+    ) {
+        let [expected_tup_elem] = &expected.tuple_fields().collect::<Vec<_>>()[..]
+            else { return };
+
+        if !same_type_modulo_infer(expected_tup_elem, found) {
+            return;
+        }
+
+        let Ok(code) = self.tcx.sess().source_map().span_to_snippet(span)
+            else { return };
+
+        const SUGGESTION: &'static str = "use a trailing comma to create a tuple with one element";
```

Comment:
No reason to use a `const` here. Also, I think `SUGGESTION` sounds like it's the suggested code.

```suggestion
        let msg = "use a trailing comma to create a tuple with one element";
```

Reply:
Makes sense to me, that's much nicer. Thanks!